encoders/numeric/CompressionVaeEncoder/__init__.py

```diff
@@ -16,68 +17,60 @@ class CompressionVaeEncoder(BaseNumericEncoder, BaseTFEncoder):
-        :param iaf_flow_length: controls how many IAF(Inverse Autoregressive Flow) flow layers the model has.
-        :param cells_encoder: determines the number, as well as size of the encoders fully connected layers.
-        :param initializer: controls how the model weights are initialized, with options being `orthogonal` (default),
-        `truncated_normal`, and `lecun_normal`
-        :param batch_size: determine the batch sizes used for training.
-        :param batch_size_test: determine the batch sizes used for testing.
-        :param feature_normalization: tells CVAE whether it should internally apply feature normalization
-        (zero mean, unit variance, based on the training data) or not. If True, the normalisation factors are stored
-        with the model and get applied to any future data.
-        :param tb_logging: determines whether the model writes summaries for TensorBoard during the training process.
+        :param model_path: specifies the path to the pretrained model
+
         """
         super().__init__(*args, **kwargs)
-        self.X = X
-        self.train_valid_split = train_valid_split
-        self.output_dim = output_dim
-        self.iaf_flow_length = iaf_flow_length
-        self.cells_encoder = cells_encoder
-        self.initializer = initializer
-        self.batch_size = batch_size
-        self.batch_size_test = batch_size_test
         self.model_path = model_path
-        self.feature_normalization = feature_normalization
-        self.tb_logging = tb_logging
 
     def post_init(self):
-        super().post_init()
-        self.to_device()
+        from cvae import cvae
+        import cvae.lib.model_iaf as model
+        import tensorflow as tf
+
+        config = tf.ConfigProto(log_device_placement=False)
+        config.gpu_options.allow_growth = True
+
+        with tf.Graph().as_default():
+            # Load parameter file.
+            with open(f'{self.model_path}/params.json', 'r') as f:
+                param = json.load(f)
+
+            net = model.VAEModel(param,
+                                 None,
+                                 input_dim=param['dim_feature'],
+                                 keep_prob=tf.placeholder_with_default(input=tf.cast(1.0, dtype=tf.float32),
+                                                                       shape=(),
+                                                                       name="KeepProb"),
+                                 initializer='orthogonal')
+            # Placeholder for data features
+            self.data_feature_placeholder = tf.placeholder_with_default(
+                input=tf.zeros([64, param['dim_feature']], dtype=tf.float32),
+                shape=[None, param['dim_feature']])
+
+            self.embeddings = net.embed(self.data_feature_placeholder)
 
-    @batching
+            self.sess = tf.Session(config=config)
+            init = tf.global_variables_initializer()
+            self.sess.run(init)
```

Comment:
Who is responsible for closing this session? Should a close be added to close it?

Reply:
We can't add a close there because in that case we are going to run the encode function on an already closed session. So the post_init can't be responsible for that. I think the session should be closed by the encode function. 
I'm going to use `with self.sess as sess:` because `with` uses a context manager which already handles the entry and exit from its block of code. So we don't need to call close().

Reply:
Yes I agree post init shouldnnot be respobsible for sure, but neither should encode since it may run more than once.

The point is about overriding `close` method of Executor. And even better but not needed to inherit ExitStack and enter context of session to have it context managed during executor lifetime, but that may not necessary.

I would just override close and ensure tjat we call parent class close and close the session.

Reply:
Now I got your point. I agree and I think it's the proper way to handle it.